nrf_security/src/mbedtls/CMakeLists.txt

```diff
@@ -184,11 +184,14 @@ zephyr_library_sources_ifdef(VANILLA_ONLY_MBEDTLS_CHACHAPOLY_C
 zephyr_library_sources_ifdef(CONFIG_MBEDTLS_ENABLE_HEAP ${NRF_SECURITY_ROOT}/src/mbedtls/mbedtls_heap.c)
 zephyr_library_app_memory(k_mbedtls_partition)
 
-if(CONFIG_SOC_NRF52840 OR CONFIG_SOC_NRF9160)
-  zephyr_library_sources(${NRF_SECURITY_ROOT}/src/backend/cc310/replacements/entropy.c)
-else()
-  zephyr_library_sources(${NRF_SECURITY_ROOT}/src/backend/nrf5x/entropy_nrf5x.c)
+if(CONFIG_ENTROPY_GENERATOR)
```

Comment:
I don't know the build system very well so which `CONFIG_ENTROPY_*` should be here is guesswork for me :-)
This does compile for me on both an nrf52840 and an nrf52810.

Please let me know if some checks here should be reordered/added/removed.